shared/src/main/scala/scala/xml/Utility.scala

```diff
@@ -243,6 +243,12 @@ object Utility extends AnyRef with parsing.TokenTests {
       }
     }
 
+  // Checks if node when converted to string is an entity ref
+  def checkNodeForEntityRef(n: Node): Boolean = {
+    val st = n.toString
+    st(0) == '&' && st(st.length - 1) == ';'
```

Comment:
do we know that `.toString` can never be empty?

also, trivial nitpick, but `st.last` is clearer than `st(st.length - 1)` imo